scripts/globals/spells/enhancing_spell.lua

```diff
@@ -318,6 +318,14 @@ xi.spells.enhancing.calculateEnhancingFinalPower = function(caster, target, spel
             finalPower = finalPower + (tier * 2)
         end
 
+        if caster:getMod(xi.mod.SHIELD_BARRIER) > 0 then
+            local shieldDefense = caster:getShieldDefense()
+
+            if shieldDefense > 0 then
+                finalPower = finalPower + shieldDefense
+            end
+        end
+
```

Comment:
```lua
        if
            caster:isPC() and
            caster:getMod(xi.mod.SHIELD_BARRIER) > 0
        then
            finalPower = finalPower + caster:getShieldDefense()
        end

Reply:
We need the PC check, so we dont clutter the console log with errors when mobs start casting protect idle.

The core functions will already make sure to check for gear and return a non nil value, so we can simplify the logic

Reply:
Also, here's some homework: What happens if a PLD subs /whm and casts protectra?
Does the other party members ALSO get the DEF bonus? Or just the PLD?

Reply:
Refactored. They do get the bonus and they should:

> A level 70+ Paladin can cast Protect V to all members of the party within range and bonus applies to all party members. 

https://www.bg-wiki.com/ffxi/Category:Protect_Spell

It should say 90+ for Protect V but it applies to all types of Protect. It also applies to the party when they use Majesty so Protectra should also get it (at least in my mind).

Reply:
Yeah, however, i did ask for protectra concretely, for a reason.

Retail treats protect and protectra as different spell families, which means the possibility is still there.

Retail is just as spaguetti as we are.

Reply:
It's free work and I'm pretty sure I have PLD over 70 let me go try to check on retail.

Reply:
Damn nope only 66.

Reply:
I can check it in retail tonight.

Better safe than sorry. One has to be nitpicky with this stuff.

Reply:
I totally understand and appreciate it!